noxfile.py

```diff
@@ -1,9 +1,33 @@
 import nox
 
+
+@nox.session(name="pip-compile", python=["3.11"])
+def pip_compile(session: nox.Session):
+    # .pip-tools.toml was introduced in v7
+    session.install("pip-tools >= 7")
```

Comment:
For reproducibitily, the `pip_compile` nox session deserves its own lockfile just like the docs one. I.e. `requirements/pip-tools.in` + `requirements/pip-tools.txt`. It's also easier to maintain the dependency definitions in a similar structre (`.in` files) vs. mixing that with specifiers hardcoded in arbitrary locations in Python scripts.